keras_nlp/models/task.py

```diff
@@ -191,7 +191,16 @@ def from_preset(
                 preset,
                 config_file="tokenizer.json",
             )
-            preprocessor = cls.preprocessor_cls(tokenizer=tokenizer)
+            if "backbone" in kwargs:
```

Comment:
Let's rework the structure of this so we don't create a useless tokenizer if a preprocessor object is getting passed in. And raise all errors first, before downloading files.

```python
if preset_cls == cls.backbone_cls:
    # Load backbone.
    if "backbone" in kwargs:
        raise ...
    backbone = load_from_preset(....

    # Load preprocessor.
    if "preprocessor" in kwargs:
        preprocessor = kwargs.pop("preprocessor")
    else:
        tokenizer = load_from_preset(...
        preprocessor = cls.preprocessor_cls(tokenizer=tokenizer)

    # Create task.
    return cls(backbone=backbone, preprocessor=preprocessor, **kwargs)
```

Reply:
Makes sense. Done.

Reply:
We should still flip the order here of this line. `raise ValueError` before `backbone = load_from_preset`.

Just a minor quality of life thing, but the `load_from_preset` call might download gigs of data and be slow, so if a user is trying to load with a custom backbone, we are better off error first, and sending the user on the right track (no need to download weights).

Reply:
Yes, that makes sense.  Done